AllReadyApp/Web-App/AllReady/wwwroot/js/index.js

```diff
@@ -1,9 +1,25 @@
-﻿(function (ko, $, navigator) {
+﻿(function (ko, $, navigator, campaigns) {
+    function Campaign(item) {
+        for (var prop in item) {
+            this[prop] = item[prop];
+        }
+        this.displayDate = function () {
+            var start = moment(this.StartDate).utcOffset(this.StartDate).format("dddd, MMMM Do YYYY");
+            var end = moment(this.EndDate).utcOffset(this.EndDate).format("dddd, MMMM Do YYYY");
+            return start + ' - ' + end;
+        }
+        return this;
+    }
 
-    var indexViewModel = {
-        status: ko.observable(),
-        activities: ko.observableArray(),
-        loadingDone: ko.observable(false),
-    };
-    ko.applyBindings(indexViewModel);
-})(ko, $, window.navigator);
\ No newline at end of file
+    function IndexViewModel(campaigns) {
+        var list = campaigns.map(function (item) { return new Campaign(item); })
+        this.campaigns = ko.observableArray(list).filterBeforeDate("EndDate").textFilter(["Name", "Description"]);
```

Comment:
None of the filter stuff is needed on `campaigns` because you aren't actually providing any filtering functionality. Exclude campaigns whose `EndDateTime` is before the current time in `HomeController`.